django/middleware/csrf.py

```diff
@@ -105,19 +105,38 @@ def _reject(self, request, reason):
         )
         return _get_failure_view()(request, reason=reason)
 
+    def _get_current_token(self, request):
+        try:
+            if settings.CSRF_USE_SESSIONS:
+                return _sanitize_token(request.session[settings.CSRF_COOKIE_NAME])
```

Comment:
I agree that considering setting `CSRF_SESSION_KEY` to a not `None` value as opt-in for enabling session based CSRF protection might be a bit confusing. It's a pattern we've not used in the past and if I remember correctly `LANGUAGE_SESSION_KEY` purpose is only to allow circumventing possible key collisions.

The other downside is that we won't be able to provide a sane default value for this key like we do with `LANGUAGE_SESSION_KEY`. While I think the approach has some merit I believe it's worth introducing two settings here.